### ISSUE TITLE
Feature Request: Add Mish activation

### ISSUE THREAD

**Issue by digantamisra98** [BrainJS/brain.js#484]
Mish is a new novel activation function proposed in this [paper](https://arxiv.org/abs/1908.08681).
It has shown promising results so far and has been adopted in several packages including:

- [TensorFlow-Addons](https://github.com/tensorflow/addons/tree/master/tensorflow_addons/activations)
- [SpaCy (Tok2Vec Layer)](https://github.com/explosion/spaCy)
- [Thinc - SpaCy's official NLP based ML library](https://github.com/explosion/thinc/releases/tag/v7.3.0)
- [Echo AI](https://github.com/digantamisra98/Echo)
- [Eclipse's deeplearning4j](https://github.com/eclipse/deeplearning4j/issues/8417)
- [Hasktorch](https://github.com/hasktorch/hasktorch/blob/e5d4fb1b11663de3e032941f78b9d25c5da06f6d/hasktorch/src/Torch/Typed/Native.hs#L314)
- [CNTKX - Extension of Microsoft's CNTK](https://github.com/delzac/cntkx)
- [FastAI-Dev](https://github.com/fastai/fastai_dev/blob/0f613ba3205990c83de9dba0c8798a9eec5452ce/dev/local/layers.py#L441)
- [Darknet](https://github.com/AlexeyAB/darknet/commit/bf8ea4183dc265ac17f7c9d939dc815269f0a213)
- [Yolov3](https://github.com/ultralytics/yolov3/commit/444a9f7099d4ff1aef12783704e3df9a8c3aa4b3)
- [BeeDNN - Library in C++](https://github.com/edeforas/BeeDNN)
- [Gen-EfficientNet-PyTorch](https://github.com/rwightman/gen-efficientnet-pytorch)
- [dnet](https://github.com/umangjpatel/dnet/blob/master/nn/activations.py)
- [DL4S](https://github.com/palle-k/DL4S)

All benchmarks, analysis and links to official package implementations can be found in this [repository](https://github.com/digantamisra98/Mish) 

It would be nice to have Mish as an option within the activation function group.

This is the comparison of Mish with other conventional activation functions in a SEResNet-50 for CIFAR-10:  
![se50_1](https://user-images.githubusercontent.com/34192716/69002745-0de37980-091b-11ea-87da-ac8d17c79e07.png)

**Comment by extremety1989** [comment]
function mish(x) {
    return x * (Math.exp(Math.log(1 + Math.exp(x))) - Math.exp(-Math.log(1 + Math.exp(x))))/(Math.exp(Math.log(1 + Math.exp(x))) + Math.exp(-Math.log(1 + Math.exp(x))));
  }


  function derivativeOfMish(y) {
    let omega = Math.exp(3 * y)+4 * Math.exp(2 * y) + (6+4 * y) * Math.exp(y) + 4 * (1 + y);
    let delta = 1 + Math.pow((Math.exp(y) + 1), 2);
    return Math.exp(y) * omega / Math.pow(delta, 2);
  }

**Comment by extremety1989** [comment]
Can you please add mish function,that i provided, i already tested it on my custom neural network and it works great,better than sigmoid and tanh ! on  XOR

**Comment by mubaidr** [comment]
That sounds great. Would love to see as contributor though.

**Comment by robertleeplummerjr** [comment]
Keep in mind we have the GPU implementations as well.

**Comment by extremety1989** [comment]
@mubaidr how can i apply this function as contributor?

**Comment by digantamisra98** [comment]
@extremety1989 are you planning to submit a PR?

**Comment by extremety1989** [comment]
@digantamisra98 no, sometimes it returns NaN when learning rate is 0.1, i do not know what is the probleme,maybe javascript

**Comment by digantamisra98** [comment]
@extremely1989 Mish has a Softplus operator which needs proper threshold to fix that NaN issue you might be facing.

**Comment by extremety1989** [comment]
@digantamisra98 my threshold is 0.5, i how much should i turn ?

**Comment by extremety1989** [comment]
@digantamisra98 
for (let i = -10; i < 10 ; i++) {
  console.log(mish(i));
}
if it is 6 it becomes NaN , i check the limit, is is 5.8

**Comment by digantamisra98** [comment]
@extremety1989 the Softplus operator thresholds that Tensorflow use is in the range of [0,20]
